hnn_core/opt_toy_example/optimize_evoked.py

```diff
@@ -0,0 +1,162 @@
+# object with no attached drives, and a dictionary of the paramters we wish to
+# optimize.
+
+
+def set_params(net, params_dict):
+
+    # Proximal 1
+    weights_ampa_p1 = {'L2_basket': params_dict['evprox1_ampa_L2_basket'],
+                       'L2_pyramidal': params_dict['evprox1_ampa_L2_pyramidal'],
+                       'L5_basket': params_dict['evprox1_ampa_L5_basket'],
+                       'L5_pyramidal': params_dict['evprox1_ampa_L5_pyramidal']}
+    synaptic_delays_p = {'L2_basket': 0.1, 'L2_pyramidal': 0.1,
+                        'L5_basket': 1., 'L5_pyramidal': 1.}
+
+    net.add_evoked_drive('evprox1',
+                         mu=params_dict['evprox1_mu'],
+                         sigma=params_dict['evprox1_sigma'],
+                         numspikes=1,
+                         location='proximal',
+                         weights_ampa=weights_ampa_p1,
+                         synaptic_delays=synaptic_delays_p)
+
+    # Distal
+    weights_ampa_d1 = {'L2_basket': params_dict['evdist1_ampa_L2_basket'],
+                       'L2_pyramidal': params_dict['evdist1_ampa_L2_pyramidal'],
+                       'L5_pyramidal': params_dict['evdist1_ampa_L5_pyramidal']}
+    weights_nmda_d1 = {'L2_basket': 0.019482, 'L2_pyramidal': 0.004317,
+                       'L5_pyramidal': 0.080074}
+    synaptic_delays_d1 = {'L2_basket': 0.1, 'L2_pyramidal': 0.1,
+                          'L5_pyramidal': 0.1}
+
+    net.add_evoked_drive('evdist1',
+                         mu=params_dict['evdist1_mu'],
+                         sigma=params_dict['evdist1_sigma'],
+                         numspikes=1,
+                         location='distal',
+                         weights_ampa=weights_ampa_d1,
+                         weights_nmda=weights_nmda_d1,
+                         synaptic_delays=synaptic_delays_d1)
+
+    # Proximal 2
+    weights_ampa_p2 = {'L2_basket': params_dict['evprox2_ampa_L2_basket'],
+                       'L2_pyramidal': params_dict['evprox2_ampa_L2_pyramidal'],
+                       'L5_basket': params_dict['evprox2_ampa_L5_basket'],
+                       'L5_pyramidal': params_dict['evprox2_ampa_L5_pyramidal']}
+
+    net.add_evoked_drive('evprox2',
+                         mu=params_dict['evprox2_mu'],
+                         sigma=params_dict['evprox2_sigma'],
+                         numspikes=1,
+                         location='proximal',
+                         weights_ampa=weights_ampa_p2,
+                         synaptic_delays=synaptic_delays_p)
+
+    return net
+
+###############################################################################
+# Then, we define the constrainst.
+#
+# The constraints must be a dictionary of tuples where the first value in each
+# tuple is the lower bound and the second value is the upper bound for the
+# corresponding parameter.
+
+
+constraints = dict()
+constraints.update({'evprox1_ampa_L2_basket': (0.01, 1.),
```

Comment:
Is this all of the parameters used in the old optimization example? My only concern is it seems like a lot of values and may take awhile to run this script

Reply:
Yup, the [old optimization example](https://jonescompneurolab.github.io/hnn-core/stable/auto_examples/howto/plot_optimize_evoked.html#sphx-glr-auto-examples-howto-plot-optimize-evoked-py) was optimizing all network drives (proximal, distal, proximal). Should we limit our example to optimizing only one drive / a few parameters from each drive?

Reply:
No just wanted to double check! Definitely can keep it as is

Reply:
Maybe we can update the old example to update only one drive / a few parameters? It will make our CIs go faster too!